### ISSUE TITLE
Feature request: Combined colour and fill aesthetic mapping

### ISSUE THREAD

**Issue by grantmcdermott** [tidyverse/ggplot2#4121]
**tl;dr** Would it be possible to add a convenience `colfill` aesthetic that controls both the colour and fill scales simultaneously?

I often find that I am mapping colour and fill scales to the exact same values (variables). When using the default **ggplot2** colour palettes and assuming no other scale transformations, this involves only a little extra typing.

``` r
library(ggplot2)

ggplot(diamonds, aes(depth, col = cut, fill = cut)) +
  geom_density(alpha = 0.3)
```

![](https://i.imgur.com/zDqSH22.png)

However, as soon as one ventures beyond the defaults, then you end up having to write out quite a lot of duplicate text which is only distinguished by the "col(our)" or "fill" prefixes. 

``` r
ggplot(diamonds, aes(depth, col = cut, fill = cut)) +
  geom_density(alpha = 0.3) +
  scale_colour_brewer(palette = 'Set2', name = 'Diamond Cut') +
  scale_fill_brewer(palette = 'Set2', name = 'Diamond Cut')
```

![](https://i.imgur.com/DuOxFl5.png)

While the above example is somewhat contrived, it's easy to think of real-life cases that would involve even more duplication. For example, if specifying custom breaks and labels, and/or a manual scale.

One potential solution would be to add a combined "colfill" aesthetic that jointly controls both colour and fill scales. Something like:

``` r
ggplot(diamonds, aes(depth, colfill = cut)) +
  geom_density(alpha = 0.3) +
  scale_colfill_brewer(palette = 'Set2', name = 'Diamond Cut') 
```

Does that seem feasible or useful?

**Comment by teunbrand** [comment]
If the aesthetics of a scale are set to both `colour` and `fill` there is a manageable amount of extra code.

``` r
library(ggplot2)
#> Warning: package 'ggplot2' was built under R version 4.0.2

ggplot(diamonds, aes(depth, col = cut, fill = cut)) +
  geom_density(alpha = 0.3) +
  scale_colour_brewer(palette = 'Set2', name = 'Diamond Cut',
                      aesthetics = c("colour",  "fill"))
```

![](https://i.imgur.com/7QWuNGH.png)

<sup>Created on 2020-07-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

What you are proposing doesn't seem to be very difficult to implement:

```
scale_colfill_brewer <- function(..., aesthetics = c("colour", "fill")) {
  scale_colour_brewer(..., aesthetics = aesthetics)
}

ggplot(diamonds, aes(depth, col = cut, fill = cut)) +
  geom_density(alpha = 0.3) +
  scale_colfill_brewer(palette = "Set2", name = "Diamond Cut")
```

**Comment by clauswilke** [comment]
That's exactly why we exposed the `aesthetics` argument in the color scales. I don't think anything else needs to be done on the ggplot2 side.
